internal/config/config.go

```diff
@@ -851,26 +852,34 @@ func loadConfig(config string) ([]byte, error) {
 }
 
 func fetchConfig(u *url.URL) ([]byte, error) {
-	req, err := http.NewRequest("GET", u.String(), nil)
+	// Read the startup flag for config-http-retry and try to convert to integer
+	configHttpRetry, err := strconv.Atoi(flag.Lookup("config-http-retry").Value.String())
 	if err != nil {
-		return nil, err
-	}
-
-	if v, exists := os.LookupEnv("INFLUX_TOKEN"); exists {
-		req.Header.Add("Authorization", "Token "+v)
-	}
-	req.Header.Add("Accept", "application/toml")
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to process config-http-retry ")
 	}
+	// Loop to retry http config request to prevent agent dying on startup if endpoint is unavailable
+	for {
+		req, err := http.NewRequest("GET", u.String(), nil)
+		if err != nil {
+			return nil, err
+		}
 
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("failed to retrieve remote config: %s", resp.Status)
+		if v, exists := os.LookupEnv("INFLUX_TOKEN"); exists {
+			req.Header.Add("Authorization", "Token "+v)
+		}
+		req.Header.Add("Accept", "application/toml")
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			log.Printf("Unable to connect to HTTP config server.  Retry in %d seconds", configHttpRetry)
+		} else if resp.StatusCode != http.StatusOK {
+			log.Printf("Error getting HTTP config, retry in %d seconds.  Status=%s", configHttpRetry, resp.Status)
+		} else {
```

Comment:
remove the use of else blocks. It indents the main-line code path and makes it harder to read. Use `continue` instead to restart the for loop.